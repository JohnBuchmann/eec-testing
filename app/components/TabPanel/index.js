import React from 'react';
import PropTypes from 'prop-types';

/**
 * TabPanel
 * @param {Object} props Props for the controller
 */
const TabPanel = (props) => {
  const { children, value, tabIndexList, ...other } = props;
  return (
    <div
      role="tabpanel"
      hidden={!tabIndexList.includes(value)}
      id={`simple-tabpanel-${tabIndexList[`${value}`]}`}
      aria-labelledby={`simple-tab-${tabIndexList[`${value}`]}`}
      {...other}
    >
      {tabIndexList.includes(value) && <>{children}</>}
    </div>
  );
};

TabPanel.propTypes = {
  children: PropTypes.node,
  tabIndexList: PropTypes.array.isRequired,
  value: PropTypes.any.isRequired,
};

export default TabPanel;
