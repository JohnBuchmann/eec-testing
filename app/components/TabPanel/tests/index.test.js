import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';

import TabPanel from '../index';
const mock = {
  value: 0,
  index: [0],
  children: 'Test',
};

describe('<TabPanel />', () => {
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <TabPanel value={mock.value} tabIndexList={mock.index}>
          {mock.children}
        </TabPanel>
      </IntlProvider>
    );
    expect(firstChild).toMatchSnapshot();
  });
});
