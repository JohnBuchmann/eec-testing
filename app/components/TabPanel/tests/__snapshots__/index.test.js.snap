// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabPanel /> should render and match the snapshot 1`] = `
<div
  aria-labelledby="simple-tab-0"
  id="simple-tabpanel-0"
  role="tabpanel"
>
  Test
</div>
`;
